core/app/modules/import-export/import.php

```diff
@@ -26,7 +31,29 @@ final public function run() {
 
 		$root_directory = new Root( $this );
 
-		return $root_directory->run_import( $manifest_data );
+		add_filter( 'elementor/document/save/data', [ $this, 'prevent_saving_elements_on_post_creation' ], 10, 2 );
+
+		$imported_posts = $root_directory->run_import( $manifest_data );
+
+		remove_filter( 'elementor/document/save/data', [ $this, 'prevent_saving_elements_on_post_creation' ], 10 );
+
+		$this->map_old_new_post_ids = $this->map_old_new_post_ids( $imported_posts );
+
+		$this->save_elements_of_imported_posts();
+
+		$this->update_object_id_of_imported_menu_items();
```

Comment:
It's may be better to pass the array to the function and not reference it from within.